src/cpp/src/llm/pipeline_stateful_npu.cpp

```diff
@@ -0,0 +1,100 @@
+
+// Copyright (C) 2025 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+
+#include "pipeline_stateful_npu.hpp"
+#include "speculative_decoding/speculative_decoding_npu.hpp"
+#include "llm/pipeline_stateful.hpp"
+#include "llm/pipeline_static.hpp"
+#include "utils.hpp"
+
+#include <fstream>
+
+#include "openvino/runtime/core.hpp"
+#include "openvino/core/parallel.hpp"
+#include "openvino/genai/text_streamer.hpp"
+
+namespace {
+    ov::genai::ModelDesc
+    extract_draft_model_from_config(ov::AnyMap& config) {
+        ov::genai::ModelDesc draft_model;
+        if (config.find(ov::genai::utils::DRAFT_MODEL_ARG_NAME) != config.end()) {
```

Comment:
Maybe some throw otherwise?

Reply:
No, no, this function returns draft model if it exists. If not, then pipeline will be switched just to `StatefulLLMPipeline`